nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -729,6 +761,17 @@ public void dump(final File dumpFile) throws IOException {
         makeRequest(DUMP_CMD, null, dumpFile, "thread dump");
     }
 
+    /**
+     * Writes NiFi status history information to the given file; if file is null, logs at
+     * INFO level instead.
+     *
+     * @param dumpFile the file to write the dump content to
+     * @throws IOException if any issues occur while writing the dump file
+     */
+    public void statusHistory(final File dumpFile, final String days) throws IOException {
```

Comment:
I think something is not quiet right. In line 246, a value is assigned to `dumpFile` explicity. (Which cannot contain null, neither because of the argument check or the `File` class's NPE check). In the other hand, the documentation of the `statusHistory` says, "if file is null", which because of this looks like an unreachable state.

Reply:
You're right, thanks for pointing out. Without a command line parser, the processing of two optional arguments along with validation resulted in excessive branching in RunNiFi, thus I opted to remove the option to write the status history dump to the bootstrap log for the moment. This might be reintroduced if a parser is added to the class.